File: app/src/main/java/org/eson/liteble/bean/BleDataBean.java
package org.eson.liteble.bean;

import java.io.Serializable;
import java.util.UUID;

/**
 * @作者 xiaoyunfei
 * @日期: 2017/3/8
 * @说明：
 */

public class BleDataBean implements Serializable {
	private static final long serialVersionUID = 3942862920612792824L;

	private String deviceAddress;
	private UUID uuid;
	private byte[] buffer;


	public BleDataBean(String deviceAddress, UUID uuid, byte[] buffer) {
		this.deviceAddress = deviceAddress;
		this.uuid = uuid;
		this.buffer = buffer;
	}

	public String getDeviceAddress() {
		return deviceAddress;
	}

	public void setDeviceAddress(String deviceAddress) {
		this.deviceAddress = deviceAddress;
	}

	public UUID getUuid() {
		return uuid;
	}

	public void setUuid(UUID uuid) {
		this.uuid = uuid;
	}

	public byte[] getBuffer() {
		return buffer;
	}

	public void setBuffer(byte[] buffer) {
		this.buffer = buffer;
	}
}

File: app/src/main/java/org/eson/liteble/detail/bean/BleDataBean.kt
package org.eson.liteble.detail.bean

import org.eson.liteble.common.util.TimeUtils
import java.util.*

/**
 * @作者 xiaoyunfei
 * @日期: 2017/3/8
 *
 * update on 2020/12/12 by xiao yun fei
 * @说明：
 */
data class BleDataBean(
        val deviceAddress: String,
        val uuid: String,
        val buffer: ByteArray,
        val time: String =TimeUtils.getCurrentTime()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as BleDataBean

        if (deviceAddress != other.deviceAddress) return false
        if (uuid != other.uuid) return false
        if (!buffer.contentEquals(other.buffer)) return false
        if (time != other.time) return false

        return true
    }

    override fun hashCode(): Int {
        var result = deviceAddress.hashCode()
        result = 31 * result + uuid.hashCode()
        result = 31 * result + buffer.contentHashCode()
        result = 31 * result + (time?.hashCode() ?: 0)
        return result
    }
}
